src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/endpoint/DataEndpoint.java

```diff
@@ -1,27 +1,200 @@
 package uk.gov.ons.ctp.integration.contactcentresvc.endpoint;
 
+import static java.util.stream.Collectors.toList;
+
 import java.util.List;
-import lombok.extern.slf4j.Slf4j;
+import java.util.UUID;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.ResponseEntity;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.RestController;
 import uk.gov.ons.ctp.integration.contactcentresvc.model.Case;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.Permission;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.PermissionType;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.Role;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.User;
 import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.CaseRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.PermissionRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.RoleRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.UserRepository;
 
-/** Example endpoint to access data from database */
-@Slf4j
+/**
+ * Example endpoints to access data from database.
+ *
+ * <p>This endpoint is purely for development use, and will be disabled or removed in future.
+ * However for now it has development value in proving the database operations.
+ *
+ * <p>This demonstrates access to the database. Note however that it responds directly with entity
+ * objects, which would not be a good idea for production code, not least because you need to avoid
+ * lazy loading issues, and also because we need separation between the API and the database
+ * internals.
+ */
 @RestController
 @RequestMapping(value = "/data", produces = "application/json")
 public class DataEndpoint {
   @Autowired CaseRepository caseRepo;
+  @Autowired UserRepository userRepo;
+  @Autowired RoleRepository roleRepo;
+  @Autowired PermissionRepository permRepo;
 
-  @RequestMapping(value = "/case", method = RequestMethod.GET)
+  @GetMapping("/case")
   public ResponseEntity<List<Case>> findCases() {
-    log.info("Entering GET findCases");
-
     List<Case> result = caseRepo.findAll();
     return ResponseEntity.ok(result);
   }
+
+  @GetMapping("/user")
```

Comment:
I have no issue with this code, but I have noticed that Phil always goes through a service layer even if its just delegating to the repo.
Im ok if Phil is ok with this

Reply:
the Class javadoc should explain the temporary nature of this code